File: kaggle_environments/envs/kore_fleets/starter_bots/ts/test/ShipyardTest.ts
import { describe } from 'mocha';
import { expect } from 'chai';

import { Shipyard } from "../kore/Shipyard";
import { Point } from '../kore/Point';

describe('Shipyard', () =>  {
    const turns = [0, 1, 2, 293, 294, 295];
    const expected = [1, 1, 2, 9, 10, 10]
    for (let i = 0; i < turns.length; i ++) {
        it(`max spawn is correct at ${turns[i]} turns controlled`, () => {
            const shipyard = new Shipyard("A", 0, new Point(0, 0), 1, turns[i], null, null);

            expect(shipyard.maxSpawn).to.equal(expected[i]);
        })
    }

});

File: kaggle_environments/envs/kore_fleets/starter_bots/java/test/ShipyardTest.java
package test;

import org.junit.Assert;
import org.junit.Test;

import kore.Point;
import kore.Shipyard;

public class ShipyardTest {

    @Test
    public void maxSpawn_worksCorrectly() {
        int[] turns = {0, 1, 2, 293, 294, 295};
        int[] expected = {1, 1, 2, 9, 10, 10};
        for (int i = 0; i < turns.length; i ++) {
            Shipyard shipyard = new Shipyard("A", 0, new Point(0, 0), 1, turns[i], null, null);

            Assert.assertEquals(shipyard.maxSpawn(), expected[i]);
        }
    }
    
}
